packages/vscode-extension/src/util/ExtensionUtil.ts

```diff
@@ -4,16 +4,24 @@
  */
 import * as vscode from 'vscode';
 
-// returns the root project folder of the active file in the editor
+// returns the root project folder of the active file in the editor if uri not defined
+// uri is present when action originated from a contextMenu of the treeView
 // works fine with workspace with multiple project folders opened
-export function getRootProjectFolder(): string | undefined {
-	const activeTextEditor = vscode.window.activeTextEditor;
-	const activeWorkspaceFolder = activeTextEditor ? vscode.workspace.getWorkspaceFolder(activeTextEditor.document.uri) : undefined;
-	return activeWorkspaceFolder ? activeWorkspaceFolder.uri.fsPath : undefined;
+export function getRootProjectFolder(uri?:vscode.Uri): string | undefined {
```

Comment:
Maybe we should move this function to a service class? I think ExtensionUtil might have been originally a place for all utility functions that are for the vscode extension. But now it looks like it's mainly used to reference dependencies coming from NodeCli.

Reply:
I copied this solution from @davidecorreu  story. I would leave it for him to change or I would change after he merge the story, to not have any merge problem.